caikit_nlp/modules/text_generation/peft_config.py

```diff
@@ -96,12 +96,128 @@ def resolve_base_model(base_model, cls, torch_dtype):
 
 
 def get_peft_config(
-    tuning_type, tuning_config, base_model, cls, torch_dtype, verbalizer
+    tuning_type,
+    tuning_config,
+    base_model,
+    cls=None,
+    torch_dtype=None,
+    verbalizer="{{input}}",
 ):
 
+    if isinstance(tuning_type, str):
+        error.value_check(
+            "<NLP65714994E>",
+            tuning_type in TuningType._member_names_,
+            f"Invalid tuning type [{tuning_type}]. Allowed types: "
+            f"[{TuningType._member_names_}]",
+        )
+        tuning_type = TuningType(tuning_type)
+
+    error.type_check("<NLP65714993E>", TuningType, tuning_type=tuning_type)
+
     if tuning_type not in TuningType._member_names_:
         raise NotImplementedError("{} tuning type not supported!".format(tuning_type))
 
+    error.type_check("<NLP65714919E>", PretrainedModelBase, base_model=base_model)
+
+    # Validate if tuned output model type is compatible with base model or not
+    output_model_types = _get_output_types(tuning_config, base_model)
+
+    # NOTE: Base model is a resource at this point
+    task_type = base_model.TASK_TYPE
+
+    error.value_check(
+        "<NLP30542004E>",
+        len(output_model_types) <= base_model.MAX_NUM_TRANSFORMERS,
+        f"Too many output model types. Got {len(output_model_types)}, "
+        f"maximum {base_model.MAX_NUM_TRANSFORMERS}",
+    )
+    # Ensure that our verbalizer is a string and will not render to a hardcoded string
+    error.value_check(
+        "<NLP83837412E>",
+        is_valid_verbalizer(verbalizer),
+        "Provided verbalizer is an invalid type or has no renderable placeholders",
+    )
+
+    # Coerce the passed model into a resource; if we have one, this is a noop
+    # TODO: When splitting up this mono-module, use the configured resource
+    #   type of the concrete class to bootstrap
+    torch_dtype = get_torch_dtype(torch_dtype)
```

Comment:
torch_dtype is not even used, didnt remove it as it would be an API breaking change, but let me know if its ok to remove here